### PR TITLE
Add traverse_ and sequence_

### LINKED ISSUE #1951: Add Traversable.sequence_ and Traversable.traverse_
```scala
def traverse_[F[_]: Applicative, A, B](ta: T[A])(f: A => F[B]): F[Unit] = traverse(ta)(f).mapConst(())
def sequence_[F[_]: Applicative, A](ta: T[F[A]]): F[Unit] = sequence(ta).mapConst(())
```
And specialize them for (at least) `IList`.

### PR DESCRIPTION
**Pull request by allantl**
Closes #1951.

@hrhino Where should I add the tests? Also, what should I do with the `FoldableSyntax` (since its using macro)?

### CODE REVIEW

File: base/shared/src/main/scala/scalaz/tc/foldable.scala [hunk 1]
```diff
@@ -16,6 +17,12 @@ trait FoldableClass[F[_]] {
   def msuml[A](fa: F[A])(implicit A: Monoid[A]): A =
     foldLeft(fa, A.mempty)(A.mappend(_, _))
 
+  def traverse_[T[_], A, B](ta: F[A])(f: A => T[B])(implicit T: Applicative[T]): T[Unit] =
+    foldLeft(ta, T.pure(()))((_, a) => T.map(f(a))(_ => ()))
```
Comment (allantl) [inline base/shared/src/main/scala/scalaz/tc/foldable.scala]: @hrhino I'd like to do `foldLeft(ta, T.pure(()))((r, a) => r <* f(a))`. 
What should I import to get `<*`?

File: base/shared/src/main/scala/scalaz/data/ilist.scala [hunk 2]
```diff
@@ -123,6 +123,12 @@ object IListModule {
             F.ap(traverse(as)(f))(f(a).map(b => (ls: IList[B]) => IList.cons(b, ls)))
         }
 
+      override def traverse_[T[_], A, B](ta: IList[A])(f: A => T[B])(implicit T: Applicative[T]): T[Unit] =
+        traverse(ta)(f).void
+
+      override def sequence_[T[_]: Applicative, A](ta: IList[T[A]]): T[Unit] =
+        traverse_(ta)(identity)
```
[the remark this answers is not in the document]
Comment (emilypi) [reply base/shared/src/main/scala/scalaz/data/ilist.scala]: LGTM

**Comment by emilypi** [comment]
@hrhino is there anything else you want out of this? I'm happy to merge if not.

**Comment by hrhino** [comment]
If you are, then I am.

Sorry for the delay, @allantl!

**Pull request merged**